src/utils/proton.js

```diff
@@ -1,31 +1,47 @@
-import { ConnectWallet } from '@protonprotocol/proton-web-sdk'
-import Foobar from '../coin.svg'
+import { ConnectWallet } from '@protonprotocol/proton-web-sdk';
+import Foobar from '../coin.svg';
 
 class ProtonSDK {
   constructor() {
-    this.chainId = '384da888112027f0321850a169f737c33e53b388aad48b5adace4bab97f437e0';
+    this.chainId =
+      '384da888112027f0321850a169f737c33e53b388aad48b5adace4bab97f437e0';
     this.endpoints = ['https://proton.greymass.com']; // Multiple for fault tolerance
     this.appName = 'Foobar';
     this.requestAccount = 'foobar'; // optional
     this.session = null;
     this.link = null;
   }
 
+  connect = async (restoreSession = false, showSelector = true) => {
+    const { link, session } = await ConnectWallet({
+      linkOptions: {
+        chainId: this.chainId,
+        endpoints: this.endpoints,
+        restoreSession,
+      },
+      transportOptions: {
+        requestAccount: this.requestAccount,
+        backButton: true,
+      },
+      selectorOptions: { appName: this.appName, appLogo: Foobar, showSelector },
+    });
+    this.link = link;
+    this.session = session;
+    return session;
```

Comment:
Since connect updates the session already, can remove the `return session` and change code after .connect() calls to use `this.session` instead of a local `session`